standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/messaging/json/JSONAlterPartitionsMessage.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.metastore.messaging.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Function;
+import com.google.common.collect.Iterables;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hive.metastore.api.Partition;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.apache.hadoop.hive.metastore.messaging.AlterPartitionsMessage;
+import org.apache.hadoop.hive.metastore.messaging.MessageBuilder;
+import org.apache.thrift.TException;
+
+public class JSONAlterPartitionsMessage extends AlterPartitionsMessage {
+
+  @JsonProperty
+  String server, servicePrincipal, db, table, tableType, tableObjJson;
+
+  @JsonProperty
+  String isTruncateOp;
+
+  @JsonProperty
+  Long timestamp, writeId;
+
+  @JsonProperty
+  List<Map<String, String>> partitions;
+
+  @JsonProperty
+  List<String> partitionListJson;
```

Comment:
The reason to serialize whole partition instead of partition name:
During replication, if the events from source cluster in db are:
```
alter_partitions
drop_table
```
When the `AlterPartitionsHandler` receives the `alter_partitions` event, if we forward to HMS to get the partition details, there should be no such objects as we have deleted the table in the source, as the `TestReplicationOptimisedBootstrap` or `TestMetaStoreEventListenerInRepl` indicates.